maggy/core/experimentdriver.py

```diff
@@ -243,6 +239,11 @@ def __init__(self, experiment_type, **kwargs):
         self.fd = hopshdfs.open_file(self.log_file, flags="w")
 
     def init(self, job_start):
+        # Set references to data in optimizer
+        self.optimizer.num_trials = self.num_trials
+        self.optimizer.searchspace = self.searchspace
+        self.optimizer.final_store = self._final_store
+        self.optimizer.direction = self.direction
```

Comment:
Why not move this block to under L252?

Reply:
No optimization object will be created if the experiment type is `ablation`, hence this will throw an exception.

Reply:
To unify this across optimization and ablation, I propose something like `AbstractController(ABC)` that contains the common attributes and objects, and `AbstractOptimizer` and `AbstractAblator` would then inherit from it. We can address this in a later PR.

Reply:
We can also implement a `ParameterServer` using an `AbstractController` - I have some ideas about this, let's discuss soon.